transport/messages/result_message.cc

```diff
@@ -22,6 +23,14 @@ std::ostream& operator<<(std::ostream& os, const result_message::bounce_to_shard
     return os;
 }
 
+std::ostream& operator<<(std::ostream& os, const result_message::exception& msg) {
+    auto [name, what] = msg.get_exception().accept([] (const auto& ex) -> std::pair<const char*, const char*> {
+        return { typeid(ex).name(), ex.what() };
+    });
+    fmt::print(os, "{{result_message::exception {} ({})}}", what);
```

Comment:
This operator<< probably shouldn't be in "result_message.cc" and should be in some general <util/outcome.hh> I suggested. Similar to how we have in Seastar's <util/log.hh>,
```
namespace std {
std::ostream& operator<<(std::ostream&, const std::exception_ptr&);
std::ostream& operator<<(std::ostream&, const std::exception&);
}
```
In your current code, maybe you can even use those existing operator<< instead of open-coding your own exception printing code.
By the way, one of the problems in your code here is that you print `typeid(ex).name()` without demangling the ugly C++-mangled names. At the minimum you should be using seastar::pretty_type_name() - but I think you can be using the entire operator<< from Seastar.

Reply:
That's a good point, I'll use the `operator<<` for the std::exception.

Reply:
Done